simulator/ns-3.35/src/point-to-point/model/switch-mmu.cc

```diff
@@ -35,39 +35,70 @@ TypeId SwitchMmu::GetTypeId(void) {
+*/
+
+
 SwitchMmu::SwitchMmu(void) {
 
-	// Buffer pools
-	bufferPool = 24 * 1024 * 1024; // ASIC buffer size i.e, total shared buffer
-	ingressPool = 18 * 1024 * 1024; // ingress lossless pool
-	egressPool[LOSSLESS] = 24 * 1024 * 1024; // egress lossless pool. Lossless bypasses egress admission
-	egressPool[LOSSY] = 14 * 1024 * 1024; // egress lossy pool.
-	egressPoolAll = 24 * 1024 * 1024;
-	xoffTotal = 0; //6 * 1024 * 1024;
+	// Here we just initalize some default values. 
+	// The buffer can be configured using Set functions through the simulation file later.
 
+	/*Buffer pools*/
+	bufferPool = 24 * 1024 * 1024; // ASIC buffer size i.e, total shared buffer
+	ingressPool = 18 * 1024 * 1024; // Size of ingress pool. Note: This is shared by both lossless and lossy traffic.
+	egressPool[LOSSLESS] = 24 * 1024 * 1024; // Size of egress lossless pool. Lossless bypasses egress admission
+	egressPool[LOSSY] = 14 * 1024 * 1024; // Size of egress lossy pool.
+	egressPoolAll = 24 * 1024 * 1024; // Not for now. For later use.
+	xoffTotal = 0; //6 * 1024 * 1024; // Total headroom space in the shared buffer pool. 
+	// xoffTotal value is incremented when SetHeadroom function is used. So setting it to zero initially. 
+	// Note: This would mean that headroom must be set explicity.
+
+
+	/*aggregate run time*/
+	/*`totalUsed` IMPORTANT TO NOTE: THIS IS NOT bytes in the "ingress pool". 
+	This is the total bytes USED in the switch buffer, which includes occupied buffer in reserved + headroom + ingresspool.*/
+	totalUsed = 0;  
+	egressPoolUsed[LOSSLESS] = 0; // Total bytes USED in the egress lossless pool
+	egressPoolUsed[LOSSY] = 0; // Total bytes USED in the egress lossy pool 
+	xoffTotalUsed = 0; // Total headroom bytes USED so far. Updated at runtime.
+	// It is sometimes useful to keep track of total bytes used specifically from ingressPool. We don't need an additional variable. 
+	// This is equal to (totalUsed - xoffTotalUsed).
 
 	for (uint32_t port = 0; port < pCnt; port++) {
 		for (uint32_t q = 0; q < qCnt; q++) {
 			// buffer configuration.
-			reserveIngress[port][q] = 1248;
-			reserveEgress[port][q] = 1248;
-			alphaEgress[port][q] = 1;
-			alphaIngress[port][q] = 1;
-			xoff[port][q] = 96928;
-			xon[port][q] = 1248;
-			xon_offset[port][q] = 2496;
-
-			// aggregate run time
-			ingressPoolUsed = 0;
-			egressPoolUsed[LOSSLESS] = 0;
-			egressPoolUsed[LOSSY] = 0;
-			totalSharedUsed = 0;
+			reserveIngress[port][q] = 1248; // Per queue reserved buffer at ingress
+			reserveEgress[port][q] = 1248; // per queue reserved buffer at egress
+			alphaEgress[port][q] = 1; // per queue alpha value used by Buffer Management/PFC Threshold at egress
+			alphaIngress[port][q] = 1; // per queue alpha value used by Buffer Management/PFC Threshold at ingress
+			xoff[port][q] = 0; // per queue headroom LIMIT at ingress. This can be changed using SetHeadroom. IMPORTANT: xoff SHOULD BE SET EXPLICITLY in a simulation.
```

Comment:
Setting `xoff` to zero initially. `xoffTotalUsed` is also set to zero initially. 

I think its best this way since the headroom depends on the port bandwidth, one hop delay etc., which are dependent on external configuration.

On the down side, it forces us to set headroom explicitly in the simulation.